src/main/java/net/sf/jabref/logic/importer/fetcher/DoiFetcher.java

```diff
@@ -0,0 +1,81 @@
+package net.sf.jabref.logic.importer.fetcher;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+
+import net.sf.jabref.logic.help.HelpFile;
+import net.sf.jabref.logic.importer.FetcherException;
+import net.sf.jabref.logic.importer.IdBasedFetcher;
+import net.sf.jabref.logic.importer.ImportFormatPreferences;
+import net.sf.jabref.logic.importer.ParserResult;
+import net.sf.jabref.logic.importer.fileformat.BibtexParser;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.logic.net.URLDownload;
+import net.sf.jabref.logic.util.DOI;
+import net.sf.jabref.model.entry.BibEntry;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public class DoiFetcher implements IdBasedFetcher {
+
+    private ImportFormatPreferences preferences;
+
+    private static final Log LOGGER = LogFactory.getLog(DoiFetcher.class);
+
+    public DoiFetcher(ImportFormatPreferences preferences) {
+        this.preferences = preferences;
+    }
+
+    public static Optional<BibEntry> getEntryFromDOI(String doi, ParserResult parserResult, ImportFormatPreferences preferences){
```

Comment:
I would prefer if this method could be removed completely. If this is not easily possible, at least mark it as deprecated.